Lib/multiprocessing/managers.py

```diff
@@ -185,15 +187,21 @@ def serve_forever(self):
                 util.debug('resetting stdout, stderr')
                 sys.stdout = sys.__stdout__
                 sys.stderr = sys.__stderr__
+            for t in self._handler_threads:
+                t.join()
+            accepter.join()
+            self.listener.close()
             sys.exit(0)
```

Comment:
Fundamentally, the accepter and handle_request threads were all being set `.daemon = True` as they were never intended to be controlled and cleaned up.  they'll all die as soon as the server process exits.  attempting to join all of the threads is perhaps _ideal_ but only works if they can be guaranteed not to be blocked on anything.  Handler threads do blocking IO to a socket connection which is not guaranteed to have been closed because it could come from an uncooperative peer.  without rewriting them to do non-blocking IO and check a stop_event themselves, joining them can't be reliable and could lead to anything making a connection to the socket being able to block cleanup.